### PR TITLE
Add Java and OS recommendations

### PR DESCRIPTION
**Pull request by aetter**
### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _opensearch/install/docker.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ docker pull opensearchproject/opensearch-dashboards:{{site.opensearch_version}}
 
 To check available versions, see [Docker Hub](https://hub.docker.com/u/opensearchproject).
 
-OpenSearch images use `centos:7` as the base image. If you run Docker locally, we recommend allowing Docker to use at least 4 GB of RAM in **Preferences** > **Resources**.
+OpenSearch images use `amazonlinux:2` as the base image. If you run Docker locally, set Docker to use at least 4 GB of RAM in **Preferences** > **Resources**.
```
Comment (aetter) [inline _opensearch/install/docker.md:19]: Hey @peterzhuamazon, this change is the result of my sleuthing, but let me know if I got it wrong.

File: _opensearch/install/compatibility.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+parent: Install OpenSearch
+nav_order: 2
+---
+
+# Operating system and JVM compatibility
+
+- We recommend installing OpenSearch on RHEL- or Debian-based Linux distributions that use [systemd](https://en.wikipedia.org/wiki/Systemd), such as CentOS, Amazon Linux 2, and Ubuntu (LTS). OpenSearch should work on many Linux distributions, but we only test a handful.
+- OpenSearch ships with a compatible version of Java in the `jdk` directory. To find its version, run `./bin/opensearch --version`.
+
+  To use a different Java installation, set the `OPENSEARCH_JAVA_HOME` environment variable to the Java install location. We recommend Java 11 (LTS), but OpenSearch also works with Java 8.
```
Comment (aetter) [inline _opensearch/install/compatibility.md:13]: It looks like our gradle build can test on a wide variety of Java versions (8-12?), but my understanding is that we only want people using the LTS ones for self-installs (8, 11, eventually 17). Again, let me know if I've got it wrong here.

File: _opensearch/install/compatibility.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+layout: default
+title: Compatibility
+parent: Install OpenSearch
+nav_order: 2
+---
+
+# Operating system and JVM compatibility
+
+- We recommend installing OpenSearch on RHEL- or Debian-based Linux distributions that use [systemd](https://en.wikipedia.org/wiki/Systemd), such as CentOS, Amazon Linux 2, and Ubuntu (LTS). OpenSearch should work on many Linux distributions, but we only test a handful.
+- The OpenSearch tarball ships with a compatible version of Java in the `jdk` directory. To find its version, run `./bin/opensearch --version`.
```
Comment (dblock) [inline _opensearch/install/compatibility.md:11]: We should say which version it is. Currently I believe it's 15.

File: _opensearch/install/compatibility.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
+layout: default
+title: Compatibility
+parent: Install OpenSearch
+nav_order: 2
+---
+
+# Operating system and JVM compatibility
```
Comment (dblock) [inline _opensearch/install/compatibility.md:8]: I think this should be split into two sections, OS and JVM

File: _opensearch/install/compatibility.md (hunk 3 above)
Comment (aetter) [reply _opensearch/install/compatibility.md]: Sure, added.

File: _opensearch/install/compatibility.md (hunk 4 above)
Comment (aetter) [reply _opensearch/install/compatibility.md:8]: This is already such a short page that I don't see a lot of utility in breaking things up further, especially if you think about it as "all the stuff you have to have in place prior to installing."

**Pull request merged**